### PR TITLE
Add VT100 ANSI escape sequences for Windows 10. Closes #3495 

### LINKED ISSUE #3495: meson color output is broken with conemu (windows)
I currently use conemu with the quake-style dropdown menu, and while trying to run meson under windows, I get the following output

screenshot 2018-04-29 00 34 23

I think that on windows, it might be best to use the [colorama](https://pypi.org/project/colorama/) library as a conditional dependency on windows. This can be done inside of the `setup.py` with the `install_requires` looking like:

```python
install_requires =
  colorama;sys_platform == 'win32'
```

Alternatively, simply importing colorama causes it to "do the right thing" on non-windows platforms. This also means you can get color support for pre-Windows 10 versions (Windows 10 added VT100 escape sequence support, but it requires an explicit opt in via a win32 function call)

I'm unsure if either of these options are valid, or even wanted.

### PR DESCRIPTION
**Pull request by slurps-mad-rips**
This change still relies on the older 'ANSICON' environment check as a fallback, in the event we're on "not Windows 10". (Calling `SetConsoleMode` with unsupported values results in a 0 being returned).

Image showing it working:

working ansi colors windows 10

This closes #3495

### CODE REVIEW

File: mesonbuild/mlog.py [hunk 1]
```diff
@@ -19,8 +19,23 @@
 information about Meson runs. Some output goes to screen,
 some to logging dir and some goes to both."""
 
+def _windows_ansi ():
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x4
+    def fallback(): return os.get.environ('ANSICON')
```
Comment (ignatenkobrain) [inline mesonbuild/mlog.py]: __[Flake8]__

[E306] expected 1 blank line before a nested definition, found 0

File: mesonbuild/mlog.py [hunk 2]
```diff
@@ -19,8 +19,23 @@
 information about Meson runs. Some output goes to screen,
 some to logging dir and some goes to both."""
 
+def _windows_ansi ():
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x4
```
Comment (ignatenkobrain) [inline mesonbuild/mlog.py]: __[Flake8]__

[F841] local variable 'ENABLE_VIRTUAL_TERMINAL_PROCESSING' is assigned to but never used

File: mesonbuild/mlog.py (hunk 1 above)
Comment (ignatenkobrain) [inline mesonbuild/mlog.py]: __[Flake8]__

[E704] multiple statements on one line (def)

File: mesonbuild/mlog.py [hunk 3]
```diff
@@ -19,8 +19,23 @@
...
+    def fallback(): return os.get.environ('ANSICON')
+    from ctypes import windll, byref
+    from ctypes.wintypes import DWORD
+
+    kernel = windll.kernel32
+    stdout = kernel.GetStdHandle(-11)
+    mode = DWORD()
+    if not kernel.GetConsoleMode(stdout, byref(mode)): return False
```
Comment (ignatenkobrain) [inline mesonbuild/mlog.py]: __[Flake8]__

[E701] multiple statements on one line (colon)

File: mesonbuild/mlog.py [hunk 4]
```diff
@@ -19,8 +19,23 @@
 information about Meson runs. Some output goes to screen,
 some to logging dir and some goes to both."""
 
+def _windows_ansi ():
```
Comment (ignatenkobrain) [inline mesonbuild/mlog.py]: __[Flake8]__

[E211] whitespace before '('

**Comment by jpakkane** [comment]
Great, thanks.

**Pull request merged**